core/src/parking_lot.rs

```diff
@@ -566,20 +603,30 @@ pub const DEFAULT_PARK_TOKEN: ParkToken = ParkToken(0);
 /// you could otherwise interfere with the operation of other synchronization
 /// primitives.
 ///
+// REVIEW: if a random key is passed in, is that the only thing that happens?
+// (messing with other primitives) If I'm the only user of `parking_lot` it's
+// memory safe to pass in any value, right?
```

Comment:
Primitives rely on not being messed with for soundness (see my comment earlier).